File: 1.0/src/c/cz/mg/vulkan/PFN_vkFlushMappedMemoryRanges.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkFlushMappedMemoryRanges_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong memoryRangeCount, jlong pMemoryRanges, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkFlushMappedMemoryRanges f = (PFN_vkFlushMappedMemoryRanges)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((uint32_t*)jniLongToPointer(memoryRangeCount)),
        ((VkMappedMemoryRange*)jniLongToPointer(pMemoryRanges))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkFlushMappedMemoryRanges.java
package cz.mg.vulkan;

public class PFNvkFlushMappedMemoryRanges extends VkFunctionPointer {
    public PFNvkFlushMappedMemoryRanges() {
    }

    protected PFNvkFlushMappedMemoryRanges(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkFlushMappedMemoryRanges(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkFlushMappedMemoryRanges(long value) {
        setValue(value);
    }

    public PFNvkFlushMappedMemoryRanges(VkInstance instance) {
        super(instance, new VkString("vkFlushMappedMemoryRanges"));
    }

    public int call(VkDevice device, int memoryRangeCount, VkMappedMemoryRange pMemoryRanges){
        return callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), memoryRangeCount, pMemoryRanges != null ? pMemoryRanges.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callNative(long vkaddress, long device, int memoryRangeCount, long pMemoryRanges);

}
